src/sql/workbench/contrib/dashboard/browser/dashboardRegistry.ts

```diff
@@ -9,17 +9,54 @@ import { IJSONSchema } from 'vs/base/common/jsonSchema';
 import * as nls from 'vs/nls';
 import { IExtensionPointUser, ExtensionsRegistry } from 'vs/workbench/services/extensions/common/extensionsRegistry';
 
-import { ProviderProperties } from 'sql/workbench/contrib/dashboard/browser/widgets/properties/propertiesWidget.component';
 import { DATABASE_DASHBOARD_TABS } from 'sql/workbench/contrib/dashboard/browser/pages/databaseDashboardPage.contribution';
 import { SERVER_DASHBOARD_TABS } from 'sql/workbench/contrib/dashboard/browser/pages/serverDashboardPage.contribution';
 import { DASHBOARD_CONFIG_ID, DASHBOARD_TABS_KEY_PROPERTY } from 'sql/workbench/contrib/dashboard/browser/pages/dashboardPageContribution';
 import { find } from 'vs/base/common/arrays';
 import { IDashboardTab, IDashboardTabGroup } from 'sql/workbench/services/dashboard/browser/common/interfaces';
+import { ILogService } from 'vs/platform/log/common/log';
+import { ServerInfo } from 'azdata';
 
 export const Extensions = {
 	DashboardContributions: 'dashboard.contributions'
 };
 
+export interface PropertiesConfig {
+	properties: Array<Property>;
+}
+
+export interface FlavorProperties {
+	flavor: string;
+	condition?: ConditionProperties;
+	conditions?: Array<ConditionProperties>;
+	databaseProperties: Array<Property>;
+	serverProperties: Array<Property>;
+	databasesListProperties?: Array<ObjectListViewProperty>;
+	objectsListProperties?: Array<ObjectListViewProperty>;
+}
+
+export interface ConditionProperties {
+	field: string;
+	operator: '==' | '<=' | '>=' | '!=';
+	value: string | boolean;
+}
+
+export interface ProviderProperties {
+	provider: string;
+	flavors: Array<FlavorProperties>;
+}
+
+export interface Property {
+	displayName: string;
+	value: string;
+	ignore?: Array<string>;
+	default?: string;
+}
+
+export interface ObjectListViewProperty extends Property {
+	widthWeight?: number;
```

Comment:
Why are you calling this widthWeight? Seems like it should just be width?

Reply:
this is actually weight. actual width of the column will be determined at runtime based on the weight.

Reply:
What do you mean "weight"? The logic looked like it was just adding up all the widths to get the total table width right? Isn't this really just a columnWidth then?

Reply:
the actual columnwidth = totalWidth * (property.widthWeight / totalColumnWidthWeight)

Reply:
is weight the right word for this property is for?